ai/core/src/unitycatalog/ai/core/databricks.py

```diff
@@ -825,3 +848,38 @@ def get_execute_function_sql_command(
         sql_query += ",".join(args)
     sql_query += ")"
     return SparkSqlCommand(sql_query=sql_query, args=params_dict)
+
+
+def dynamically_construct_python_function(function_info: "FunctionInfo") -> str:
+    """
+    Construct a Python function from the given FunctionInfo object.
+
+    Note: This function will not recreate the original docstring of the function.
+
+    Args:
+        function_info: The FunctionInfo object containing the function metadata.
+
+    Returns:
+        The re-constructed function definition as a string.
+    """
+    param_names = []
+    if function_info.input_params and function_info.input_params.parameters:
+        for param in function_info.input_params.parameters:
+            param_names.append(extract_collection_types(param))
+    return_type = parse_full_sql_data_type_for_return_type(function_info.full_data_type)
+    function_head = f"{function_info.name}({', '.join(param_names)}) -> {return_type}"
+    func_def = f"def {function_head}:\n"
+
+    docstring = reconstruct_docstring(function_info)
+    if docstring:
+        func_def += docstring
+
+    if function_info.routine_body.value == "EXTERNAL":
```

Comment:
Let's move this check up so we fail faster?

Reply:
Added in #912 :)